lib/srv/db/common/packetcapture/capture.go

```diff
@@ -0,0 +1,248 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package packetcapture provides utilities for saving application-layer packets to file in either plain text or PCAP formats.
+// The PCAP functionality depends on the external utilities from Wireshark (text2pcap, mergecap) and is expected to be used in dev/debugging contexts only.
+package packetcapture
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+)
+
+// Direction represents the link and participants.
+type Direction int
+
+const (
+	ClientToTeleport Direction = iota
+	ServerToTeleport
+	TeleportToClient
+	TeleportToServer
+)
+
+const fakeClientAddr = "1.1.1.1"
+const fakeTeleportAddr = "2.2.2.2"
+const fakeServerAddr = "3.3.3.3"
+
+const mergecapBin = "mergecap"
+const text2pcapBin = "text2pcap"
+
+func (p Direction) String() string {
+	switch p {
+	case ClientToTeleport:
+		return "Client->Teleport"
+	case ServerToTeleport:
+		return "Server->Teleport"
+	case TeleportToClient:
+		return "Teleport->Client"
+	case TeleportToServer:
+		return "Teleport->Server"
+	default:
+		return fmt.Sprintf("Unknown(%d)", p)
+	}
+}
+
+// PacketEntry holds the details of each packet to be recorded.
+type PacketEntry struct {
+	Direction Direction
+	Payload   []byte
+	Timestamp time.Time
+}
+
+// Capture struct holds all packet details and offers methods to add and save packets.
+type Capture struct {
+	packets    []PacketEntry
+	clock      clockwork.Clock
+	runCommand func(name string, arg ...string) ([]byte, error)
```

Comment:
Add a godoc hint that this is expected to behave like `(*exec.Cmd).CombinedOutput`?